packages/navie/src/context.ts

```diff
@@ -1,19 +1,84 @@
-export type ContextRequest = {
-  type: 'search';
-  vectorTerms: string[];
-  tokenCount: number;
-};
-
-export type ContextItem = {
-  name: string;
-  score?: number;
-  content: string;
-};
-
-export type ContextResponse = {
-  sequenceDiagrams: string[];
-  codeSnippets: { [key: string]: string };
-  codeObjects: string[];
-};
-
-export type ContextProvider = (request: ContextRequest) => Promise<ContextResponse>;
+/* eslint-disable @typescript-eslint/no-namespace */
+export namespace ContextV1 {
+  export type ContextRequest = {
+    vectorTerms: string[];
+    tokenCount: number;
+  };
+
+  export type ContextItem = {
+    name: string;
+    score?: number;
+    content: string;
+  };
+
+  export type ContextResponse = {
+    sequenceDiagrams: string[];
+    codeSnippets: { [key: string]: string };
+    codeObjects: string[];
+  };
+
+  export type ContextProvider = (
+    request: ContextRequest & { version: 1; type: 'search' }
+  ) => Promise<ContextResponse>;
+}
+
+export namespace ContextV2 {
+  // Describes the different types of context data that is available.
+  export enum ContextItemType {
+    // A summarized sequence diagram that describes the overall flow of the application.
+    // The diagram is synthesized from a trace of the application's execution.
+    SequenceDiagram = 'sequence-diagram',
+    // A code snippet that is relevant to the context. The snippet is a few lines
+    // from a source file that is related to the context.
+    CodeSnippet = 'code-snippet',
+    // A data request that was made by the application. This could be a database query,
+    // an HTTP client request, or any other type of data request.
+    DataRequest = 'data-request',
+  }
+
+  // A specific context item that is returned in the response.
+  export type ContextItem = {
+    // Identifies the location in the code base from which the context was obtained.
+    // For example, a file path with an optional line number or range in the format "path/to/file.rb:1-2".
+    location?: string;
+    // The type of context item.
+    type: ContextItemType;
+    // The content of the context item. The content is specific to the type of context item.
+    content: string;
+    // A score that indicates the relevance of the context item to the request. Score is present
+    // only when the context item was obtained by searching for keywords (terms).
+    score?: number;
+  };
+
+  // Request a set of context items from the context provider.
+  export type ContextRequest = {
```

Comment:
TODO: Some of these fields, e.g. `recent` are for future use, so they should be removed for now since the context provider does not implement them yet.